src/java.net.http/share/classes/jdk/internal/net/http/RequestPublishers.java

```diff
@@ -202,13 +201,13 @@ public long contentLength() {
 
     public static class StringPublisher extends ByteArrayPublisher {
         public StringPublisher(String content, Charset charset) {
-            super(content.getBytes(charset));
+            super(content.getBytes(Objects.requireNonNull(charset)));   // Implicit null check on `content`
         }
     }
 
     public static class EmptyPublisher implements BodyPublisher {
```

Comment:
```suggestion
    public static final class EmptyPublisher implements BodyPublisher {
```

Reply:
I kept unrelated improvements limited to the lines already touched, to avoid cluttering the diff and the VCS history. While I fully agree with your proposal, I'd prefer to address it some other time.